feat(cu_worker)!: send stats over gRPC

monoid opened the pull request:
Breaking change: gRPC definitions are changed, a `report_proof` method is renamed to `report`, new fields are added.

Commit:
feat(cu_worker): cpu worker stats to console

Commit:
Fix typo

Commit:
Fix crossterm feature

Commit:
Protobuf stats sketch

1. The `StatsRequest` contains number of `HashRate` records (in case of
   connectivity problems).
2. Each `HashRate` contains both counts and durations; any of them
   intended to be zero if unrelevant.

Commit:
Send stat records to gRPC endpoint

mikevoronov commented on crates/cu-worker/src/metrics.rs:
```diff
@@ -0,0 +1,41 @@
+use ccp::stats_collecting_thread::StatsSink;
+use ccp_cu::proving_thread::sync::to_stats_collecting_message::{
+    ToStatsCollectingInlet, ToStatsCollectingMessage, ToStatsCollectingOutlet,
+};
```
decouple, new line after

mikevoronov reviewed (approved):
utACK + comments in #196.

Commit:
Merge branch 'main' into feat/rpc-stats

Commit:
WIP: send EpochParameters with stats

It allows lightmare to filter out stale epoch's records.

Commit:
Fine-tune some gRPC code

Commit:
Magic constant

Commit:
Revert "WIP: send EpochParameters with stats"

This reverts commit bd304fd9e066da20a78176353171a8de442fd1a3.

Commit:
Epoch parameters in the gRPC stats

Store epoch parameters `ToStatsCollectingMessage` to simplify the stats
handling code.
